File: reflex-wasm/src/imports.mjs
// SPDX-FileCopyrightText: 2023 Marshall Wace <opensource@mwam.com>
// SPDX-License-Identifier: Apache-2.0
// SPDX-FileContributor: Tim Kendrick <t.kendrick@mwam.com> https://github.com/timkendrickmw
export default function imports(wasi) {
  return {
    Math: {
      remainder: (x, y) => x % y,
      acos: Math.acos,
      acosh: Math.acosh,
      asin: Math.asin,
      asinh: Math.asinh,
      atan: Math.atan,
      atan2: Math.atan2,
      atanh: Math.atanh,
      cbrt: Math.cbrt,
      cos: Math.cos,
      cosh: Math.cosh,
      exp: Math.exp,
      expm1: Math.expm1,
      hypot: Math.hypot,
      log: Math.log,
      log2: Math.log2,
      log10: Math.log10,
      log1p: Math.log1p,
      pow: Math.pow,
      sin: Math.sin,
      sinh: Math.sinh,
      sqrt: Math.sqrt,
      tan: Math.tan,
      tanh: Math.tanh,
    },
    wasi_snapshot_preview1: wasi.wasiImport,
  };
}

File: reflex-graphql/src/imports.rs
// SPDX-FileCopyrightText: 2023 Marshall Wace <opensource@mwam.com>
// SPDX-License-Identifier: Apache-2.0
// SPDX-FileContributor: Tim Kendrick <t.kendrick@mwam.com> https://github.com/timkendrickmw
use reflex::core::{Builtin, Expression, ExpressionFactory, HeapAllocator};

use crate::{imports::graphql::import_graphql, stdlib::GraphQlResolver};

pub mod graphql;

pub trait GraphQlImportsBuiltin: Builtin + From<GraphQlResolver> {}
impl<T> GraphQlImportsBuiltin for T where T: Builtin + From<GraphQlResolver> {}

pub fn graphql_imports<T: Expression>(
    factory: &impl ExpressionFactory<T>,
    allocator: &impl HeapAllocator<T>,
) -> Vec<(String, T)>
where
    T::Builtin: GraphQlImportsBuiltin,
{
    vec![(
        String::from("reflex::graphql"),
        import_graphql(factory, allocator),
    )]
}
